server/sonar-server/src/test/java/org/sonar/server/usertoken/ws/GenerateActionTest.java

```diff
@@ -93,6 +97,15 @@ public void return_json_example() {
   }
 
   @Test
+  public void a_user_can_generate_token_for_himself() {
+    userSession.login(GRACE_HOPPER).setGlobalPermissions(GlobalPermissions.SCAN_EXECUTION);
+
+    GenerateWsResponse response = newRequest(null, TOKEN_NAME);
+
+    assertThat(response.getLogin()).isEqualTo(GRACE_HOPPER);
+  }
+
+  @Test
   public void fail_if_login_does_not_exist() {
     expectedException.expect(NotFoundException.class);
     expectedException.expectMessage("User with login 'unknown-login' not found");
```

Comment:
Error message of `POST api/user_tokens/generate?name=test` with anonymous user could be improved:
```
{"errors":[{"msg":"User with login 'null' not found"}]}
```